akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/Running.scala

```diff
@@ -110,6 +111,57 @@ private[akka] object Running {
       applyEffects(cmd, state, effect.asInstanceOf[EffectImpl[E, S]]) // TODO can we avoid the cast?
     }
 
+    /** Handle journal responses for non-persist events workloads. */
+    def onJournalResponse(response: JournalProtocol.Response): Behavior[InternalProtocol] = {
+      setup.log.debug("Received Journal response: {}", response)
+
+      val signal = response match {
+        case e @ DeleteMessagesSuccess(toSequenceNr) =>
+          setup.log.debug("PersistentShard messages to [{}] deleted successfully.", toSequenceNr)
+          internalDeleteSnapshots(e)
+          Some(DeleteMessagesCompleted(toSequenceNr))
+        case DeleteMessagesFailure(e, toSequenceNr) =>
+          setup.log.warning("Failed to delete messages toSequenceNr [{}] for persistenceId [{}] due to [{}: {}].",
+            toSequenceNr, setup.persistenceId.id, e.getClass.getCanonicalName, e.getMessage)
+          Some(DeleteMessagesFailed(toSequenceNr, e))
+        case _ =>
+          None
+      }
+
+      signal.foreach(setup.onSignal _)
+      this
+    }
+
+    /** Handle snapshot responses for non-persist events workloads. */
+    def onSnapshotterResponse(response: SnapshotProtocol.Response): Behavior[InternalProtocol] = {
+      setup.log.debug("Received Snapshotter response {}", response)
+
+      val signal = response match {
+        case DeleteSnapshotsSuccess(criteria) =>
+          setup.log.debug(s"Persistent snapshots [{}] deleted successfully", criteria)
+          Some(DeleteSnapshotsCompleted(DeletionTarget.Criteria(criteria)))
+
+        case DeleteSnapshotsFailure(criteria, error) =>
+          setup.log.warning("Failed to delete snapshots given criteria [{}] due to: [{}: {}]",
+            criteria, error.getClass.getCanonicalName, error.getMessage)
+          Some(DeleteSnapshotsFailed(DeletionTarget.Criteria(criteria), error))
+
+        case DeleteSnapshotSuccess(meta) =>
+          setup.log.debug(s"Persistent snapshot [{}] saved successfully", meta)
+          Some(DeleteSnapshotsCompleted(DeletionTarget.Individual(meta)))
+
+        case DeleteSnapshotFailure(meta, error) =>
+          setup.log.debug("Failed to delete snapshot with meta [{}] due to: [{}]", meta, error)
+          Some(DeleteSnapshotsFailed(DeletionTarget.Individual(meta), error))
+
```

Comment:
Do we really need the `DeleteSnapshotSuccess`, `DeleteSnapshotFailure`

Reply:
If we do keep it I don't think we should log here, it should be in the default signal handler so that if the user does catch the signal and provide their own logic they override the default (logging).